.github/workflows/beam_PreCommit_Java_Flink.yml

```diff
@@ -0,0 +1,94 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Java Flink Versions PreCommit
+
+on:
+  push:
+    tags: ['v*']
+    branches: ['master', 'release-*']
+    paths:
+      - 'model/**'
+      - 'runners/flink/**'
+      - 'release/**'
+      - '.github/workflows/beam_PreCommit_Java_Flink_Versions.yml'
+  pull_request_target:
+    branches: ['master', 'release-*']
+    paths:
+      - 'model/**'
+      - 'runners/flink/**'
+      - 'release/**'
+  issue_comment:
+    types: [created]
+  schedule:
+  - cron: '20 */6 * * *'
+  workflow_dispatch:
+
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.sha || github.head_ref || github.ref }}-${{ github.event.sender.login }}-${{ github.event.schedule }}'
+  cancel-in-progress: true
+
+jobs:
+  beam_PreCommit_Java_Flink_Versions:
```

Comment:
We also need to add the `timeout-minutes:` limit (see the [example](https://github.com/apache/beam/pull/27712/files#diff-4111d33e8e4af0c7fbee46a4db7f83ba0a2ac4f206decb441bfc07fa4e13d6adR56)). The default GH limit is `360`, but the default limit that is used for jenkins jobs is `120` - https://github.com/apache/beam/blob/master/.test-infra/jenkins/PrecommitJobBuilder.groovy#L39. We should use this to keep track of whether we have a performance regression or not.

Reply:
set to 120 min